app/assets/javascripts/discourse/app/components/dialog-messages/second-factor-confirm-phrase.js

```diff
@@ -0,0 +1,22 @@
+import Component from "@glimmer/component";
+import I18n from "I18n";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+
+export default class SecondFactorConfirmPhrase extends Component {
+  @service dialog;
+  @service currentUser;
+
+  @tracked confirmPhraseInput = "";
+  disabledString = I18n.t("user.second_factor.disable");
+
+  @action
+  onConfirmPhraseInput() {
+    if (this.confirmPhraseInput === this.disabledString) {
+      this.dialog.set("confirmButtonDisabled", false);
```

Comment:
Not related to this PR, but generally it would be great to octane-ify the dialog service to avoid the need for `.set()`

Reply:
Yes, agreed, beyond the scope of this PR (but certainly a nice, somewhat small task for a separate PR).